esp/src/src/Utility.ts

```diff
@@ -905,3 +905,65 @@ export class Persist {
 export function textColor(backgroundColor: string): string {
     return Palette.textColor(backgroundColor);
 }
+
+function toCSVCell(str) {
+    str = "" + str;
+    const mustQuote = (str.indexOf(",") >= 0 || str.indexOf("\"") >= 0 || str.indexOf("\r") >= 0 || str.indexOf("\n") >= 0);
+    if (mustQuote) {
+        let retVal = "\"";
+        for (let i = 0; i < str.length; ++i) {
+            const c = str.charAt(i);
+            retVal += c === "\"" ? "\"\"" : c;
+
+        }
+        retVal += "\"";
+        return retVal;
+    }
+    return str;
+}
+
+function csvFormatHeader(data, delim) {
+    let retVal = "";
+    if (data.length) {
+        for (const key in data[0]) {
+            if (retVal.length)
+                retVal += delim;
+            retVal += key;
+        }
+    }
+    return retVal;
+}
+
+function csvFormatRow(row, idx, delim) {
+    let retVal = "";
+    for (const key in row) {
+        if (retVal.length)
+            retVal += delim;
+        retVal += toCSVCell(row[key]);
+    }
+    return retVal;
+}
+
+function csvFormatFooter(data) {
+    return "";
+}
+
+export function toCSV(data, delim) {
```

Comment:
Might want to default the delim to "," since it's "CSV"

Reply:
Yup - changes made as suggested.